### PR TITLE
SDK-1309 add task set sub-tasks relations

### PR DESCRIPTION
**Pull request by sergeyspryker**
## Issue ticket link
Ticket:[SDK-1309](https://spryker.atlassian.net/browse/SDK-1309)

## Checklist before requesting a review
- [x] I have added ticket link to the PR description.
- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: src/Core/Application/Dependency/Repository/TaskSetTaskRelationRepositoryInterface.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Core\Application\Dependency\Repository;
+
+use SprykerSdk\Sdk\Core\Domain\Entity\TaskSetTaskRelationInterface;
+
+interface TaskSetTaskRelationRepositoryInterface
```
Comment (vol4onok) [inline src/Core/Application/Dependency/Repository/TaskSetTaskRelationRepositoryInterface.php:12]: Do we need this repo for one-to-many? Can we resolve it with annotations in task model?

File: src/Infrastructure/Mapper/TaskSetTaskRelationMapper.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+use SprykerSdk\Sdk\Infrastructure\Entity\TaskSetTaskRelation as InfrastructureTaskSetRelation;
+
+class TaskSetTaskRelationMapper implements TaskSetTaskRelationMapperInterface
+{
+    /**
+     * @var \Doctrine\Persistence\ObjectRepository<\SprykerSdk\Sdk\Infrastructure\Entity\Task>
+     */
+    protected ObjectRepository $taskRepository;
+
+    protected TaskMapperInterface $taskMapper;
```
Comment (vol4onok) [inline src/Infrastructure/Mapper/TaskSetTaskRelationMapper.php:23]: annotation

File: src/Presentation/Console/Command/TaskRunFactoryLoader.php [hunk 3]
```diff
@@ -64,6 +65,8 @@ class TaskRunFactoryLoader extends ContainerCommandLoader
      */
     protected ContextFactoryInterface $contextFactory;
 
+    protected TaskSetTaskRelationRepositoryInterface $taskSetTaskRelationRepository;
```
Comment (vol4onok) [inline src/Presentation/Console/Command/TaskRunFactoryLoader.php:68]: annotation

File: src/Infrastructure/Repository/TaskSetTaskRelationRepository.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+use SprykerSdk\Sdk\Core\Domain\Entity\TaskSetTaskRelationInterface;
+use SprykerSdk\Sdk\Infrastructure\Entity\TaskSetTaskRelation;
+use SprykerSdk\Sdk\Infrastructure\Mapper\TaskSetTaskRelationMapperInterface;
+
+/**
+ * @extends \Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository<\SprykerSdk\Sdk\Infrastructure\Entity\TaskSetTaskRelation>
+ */
+class TaskSetTaskRelationRepository extends ServiceEntityRepository implements TaskSetTaskRelationRepositoryInterface
+{
+    protected TaskSetTaskRelationMapperInterface $taskSetTaskRelationMapper;
```
Comment (vol4onok) [inline src/Infrastructure/Repository/TaskSetTaskRelationRepository.php:22]: annotation

**Review by vol4onok — APPROVED** [review]

File: tests/Sdk/Unit/Infrastructure/Builder/TaskSet/TaskSetTaskRelationsBuilderTest.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+     * @return void
+     */
+    public function testBuildFromTaskSetShouldThrowExceptionWhenSubTaskIdNotInExistingTasks(): void
+    {
+        // Arrange
+        $taskSetMock = $this->createTaskSetMock('task-set:id', ['task:id']);
+        $taskSetTaskRelationsBuilder = new TaskSetTaskRelationsBuilder();
+        $existingTasks = ['task-set:id' => $taskSetMock];
+
+        $this->expectException(InvalidArgumentException::class);
```
Comment (pavelmaksimov25) [inline tests/Sdk/Unit/Infrastructure/Builder/TaskSet/TaskSetTaskRelationsBuilderTest.php:53]: // Assert

The same for the method above and the rest of new tests with the exception checks.

**Review by pavelmaksimov25 — CHANGES_REQUESTED** [review]
Please provide unit tests for the `Mapper` and acceptance test to check the decorated help message in the output.

File: src/Presentation/Console/Command/TaskRunFactoryLoader.php (hunk 3 above)
Comment (vol4onok) [reply src/Presentation/Console/Command/TaskRunFactoryLoader.php:71]: probably yes. CS sometime passes some issues.

**Review by pavelmaksimov25 — DISMISSED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Pull request merged**